src/client.rs

```diff
@@ -153,6 +153,14 @@ pub struct RequestBuilder {
 }
 
 impl RequestBuilder {
+
+    /// Get the url result.
+    ///
+    /// Will contain an error if the given URL was invalid.
+    pub fn get_url(&self) -> &Result<Url, ::UrlError> {
```

Comment:
The alternative would be returning a `Result<RequestBuilder>` from all the client methods.

Which would probably be the clearer way to do it anyway.
I'd prefer to get invalid Url error early rather then when sending.